src/explore-education-statistics-frontend/src/modules/find-statistics/components/PublicationList.tsx

```diff
@@ -1,58 +1,100 @@
 import { PublicationSummary } from '@common/services/themeService';
+import {
+  PublicationType,
+  publicationTypes,
+} from '@common/services/types/publicationType';
 import Link from '@frontend/components/Link';
 import React from 'react';
 
 interface Props {
   publications: PublicationSummary[];
 }
 
+const publicationTypeReferenceOrder: PublicationType[] = [
+  'NationalAndOfficial',
+  'Experimental',
+  'AdHoc',
+  'ManagementInformation',
+  'Legacy',
+];
+
+const groupPublicationsByType = (publications: PublicationSummary[]) =>
+  Object.entries(
+    publications.reduce((acc, publication) => {
+      (acc[publication.type] ??= []).push(publication);
+      return acc;
+    }, {} as { [key in PublicationType]?: PublicationSummary[] }),
```

Comment:
Minor thing really, but realized this can be simplified using Lodash's `groupBy`:

```ts
import groupBy from 'lodash/groupBy';

Object.entries(groupBy(publications, publication => publication.type))
```

Side note whilst we're here - always import the specific Lodash function:

```ts
import { groupBy } from 'lodash'; ❌
import groupBy from 'lodash/groupBy'; ✅
```

The former imports the entirety of Lodash and would increase the final bundle size considerably.

Reply:
I originally used Lodash but switched to the native reduce thinking it would assist getting the typed keys rather than strings. I found this didn't work out but decided to stick with the native reduce rather than importing the Lodash function.

I've switched to using Lodash now to simplify things, thanks.